### PR TITLE
Upgrade from Chromium 122 to Chromium 123

### PR DESCRIPTION
**Pull request by emerick**
Resolves https://github.com/brave/brave-browser/issues/35807

Also:

Resolves https://github.com/brave/brave-browser/issues/36651

## Submitter Checklist:

- [ ] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

### CODE REVIEW

**Review by zenparsing — APPROVED** [review]
Rewards 👍

**Review by tmancey — APPROVED** [review]
LGTM++

**Review by StephenHeaps — APPROVED** [review]
iOS wallet++

**Review by yrliou — APPROVED** [review]
++ for wallet, ipfs, filecoin, brave_service_key_utils, ai_chat

**Review by rillian — APPROVED** [review]
rust++

Vendors bitflags 1.3.2 now that chromium has moved to v2. That's fine; we can migrate as we update other crates.

Adds a number of patches, which we'll have to take care of when updating later. Fortunately they all look like changes which will trigger build failures if they're dropped, so there risk of regression is low there.

**Review by DJAndries — APPROVED** [review]
p3a++

**Review by simonhong — APPROVED** [review]
`browser/theme` ++
`browser/ui/views/infobars/brave_wayback_machine_*` ++

**Review by AlexeyBarabash — APPROVED** [review]
sync++; android <=> desktop basic check passed

File: ios/browser/svg/BUILD.gn [hunk 1]
```diff
@@ -57,6 +57,9 @@ source_set("svg") {
   sources += skia_svg_public
   sources += skia_svg_sources
 
+  sources -=
+      [ "//third_party/skia/modules/svg/src/SkSVGOpenTypeSVGDecoder.cpp" ]
+
```
Comment (kylehickinson) [inline ios/browser/svg/BUILD.gn:62]: Not sure if related but may want to keep an eye on https://github.com/brave/brave-core/pull/22434/files when rebasing

**Review by darkdh — APPROVED** [review]

File: browser/ui/views/sidebar/sidebar_items_scroll_view.cc [hunk 2]
```diff
@@ -217,7 +217,7 @@ void SidebarItemsScrollView::OnItemAdded(const sidebar::SidebarItem& item,
 
   // Calculate and set this view's bounds to determine whether this view is
   // scroll mode or not.
-  parent()->Layout();
+  DeprecatedLayoutImmediately();
```
Comment (simonhong) [inline browser/ui/views/sidebar/sidebar_items_scroll_view.cc:220]:
```suggestion
  parent()->DeprecatedLayoutImmediately();
```

Comment (emerick) [reply browser/ui/views/sidebar/sidebar_items_scroll_view.cc]: Thanks for finding that @simonhong, fixed here 106e3eb78c39cbb78f19a11d6d99937d76859eca (and workaround removed).

File: browser/ui/commander/fuzzy_finder.cc [hunk 3]
```diff
@@ -0,0 +1,455 @@
...
+  base::ranges::reverse(positions);
+  ConvertPositionsToRanges(positions, matched_ranges);
+  return max_score;
+}
+
+}  // namespace
+
+namespace commander {
+
+FuzzyFinder::FuzzyFinder(const std::u16string& needle)
```
Comment (goodov) [inline browser/ui/commander/fuzzy_finder.cc:235]: there's a lot of stuff going on here. Hopefully this will not be a support burden.

**Review by goodov — APPROVED** [review]

File: browser/ui/commander/fuzzy_finder.cc (hunk 3 above)
Comment (cdesouza-chromium) [reply browser/ui/commander/fuzzy_finder.cc:235]: cc @fallaciousreasoning.

**Review by bsclifton — APPROVED** [review]
js_deps and string reviewer ++

**Review by iefremov — APPROVED** [review]

**Review by fallaciousreasoning — APPROVED** [review]
Commander changes LGTM

**Comment by simonhong** [comment]
Sorry for the inconvenient. https://github.com/brave/brave-core/pull/22489/files made patch file conflict.

**Comment by bsclifton** [comment]
Things look good here - will be doing an admin merge 😄

**Comment by bsclifton** [comment]
Test that failed is flakey - will address in follow up by disabling. Merging

**Pull request merged**